experimental/entity/generic.go

```diff
@@ -0,0 +1,129 @@
+package entity
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+type GenericEntity struct {
+	Envelope
+
+	Body interface{} `json:"body,omitempty"`
+}
+
+var _ Kind = &GenericKind{}
+
+type GenericKind struct {
+	info KindInfo
+}
+
+func NewGenericKind(info KindInfo) *GenericKind {
+	return &GenericKind{info: info}
+}
+
+func (k *GenericKind) Info() KindInfo {
+	return k.info
+}
+
+func (k *GenericKind) GoType() interface{} {
+	return &GenericEntity{}
+}
+
+func (k *GenericKind) Read(payload []byte) (interface{}, error) {
+	g := &GenericEntity{}
+	err := json.Unmarshal(payload, g)
+	if err != nil {
+		return nil, err
+	}
+	if g.Kind == "" {
+		g.Kind = k.info.ID
+	} else if g.Kind != k.info.ID {
+		return nil, fmt.Errorf("expected kind: %s", k.info.ID)
+	}
+	return g, nil
+}
+
+func (k *GenericKind) Write(v interface{}) ([]byte, error) {
+	g, ok := v.(*GenericEntity)
+	if !ok {
+		return nil, fmt.Errorf("expected RawFileEntity")
+	}
+	return json.MarshalIndent(g, "", "  ")
+}
+
+func (k *GenericKind) GetReferences(v interface{}) []EntityLocator {
+	return nil
+}
+
+func (k *GenericKind) Normalize(payload []byte, details bool) NormalizeResponse {
```

Comment:
I think both this and the `Migrate` method need to return `(entity.NormalizeResponse, error)` rather than just the `entity.NormalizeResponse` to differentiate between random, transient system errors (e.g. sanitization API timeout) and persistent entity validation failures